### PR TITLE
make Generator.generate partial

### PR DESCRIPTION
**Pull request by mfelsche**
and ignore failures up to a configurable number of retries..

So Generators can be more easily created for partial operations.

Also change some Generator combinators and factory methods, most importantly 'one_of' and 'frequency' to not be partial anymore themselves.

### CODE REVIEW

**Comment by Theodus** [comment]
There are conflicts, but they seem straightforward to fix. I'll try to review this tomorrow

**Comment by mfelsche** [comment]
conflicts resolved

File: examples/custom_class.pony [hunk 1]
```diff
@@ -129,16 +120,13 @@ class _CustomClassCustomGeneratorProperty is Property1[MyLittlePony]
...
-        fun generate(rnd: Randomness): GenerateResult[MyLittlePony] =>
-          (let name, let name_shrinks) = name_gen.generate_and_shrink(rnd)
-          (let cuteness, let cuteness_shrinks) = cuteness_gen.generate_and_shrink(rnd)
-          (let color, let color_shrinks) = color_gen.generate_and_shrink(rnd)
+          Generators.one_of[Color]([Blue; Green; Pink; Rose] where do_shrink=true)
+
+        fun generate(rnd: Randomness): GenerateResult[MyLittlePony] ? =>
+          (let name, let name_shrinks) = name_gen.generate_and_shrink(rnd)?
+          (let cuteness, let cuteness_shrinks) =
+          cuteness_gen.generate_and_shrink(rnd)?
```
Comment (Theodus) [inline examples/custom_class.pony]: This line should be indented

File: ponycheck/test/gen.pony [hunk 2]
```diff
@@ -72,34 +72,70 @@ class GenUnionTest is UnitTest
...
-      for i in Range(0, 100) do
-        generated.push(gen.generate_value(rnd))
-      end
-      h.assert_false(generated.contains(U8(42)), "frequency generated value with 0 weight")
-      h.assert_true(generated.contains(U8(0)), "frequency did not generate value with weight of 1")
-      h.assert_true(generated.contains(U8(1)), "frequency did not generate value with weight of 2")
-    else
-      h.fail("error creating frequency generator")
+    let gen = Generators.frequency[U8]([
+       as WeightedGenerator[U8]:
```
Comment (Theodus) [inline ponycheck/test/gen.pony]: I think this indentation is off by a space

**Review by Theodus — CHANGES_REQUESTED** [review]
Looks good, other than a few formatting concerns

**Comment by mfelsche** [comment]
i formatted the mentioned lines. assuming this is good, i am gonna merge.

**Pull request merged**